ovirtlago/virt.py

```diff
@@ -118,34 +118,35 @@ def wait_for_ssh(self):
 class EngineVM(lago.vm.DefaultVM):
     def __init__(self, *args, **kwargs):
         super(EngineVM, self).__init__(*args, **kwargs)
-        self._api = None
+        self._api_v3 = None
 
     def stop(self):
         super(EngineVM, self).stop()
-        self._api = None
+        self._api_v3 = None
 
     def _artifact_paths(self):
         inherited_artifacts = super(EngineVM, self)._artifact_paths()
         return set(inherited_artifacts + ['/var/log/ovirt-engine', ])
 
-    def _create_api(self):
+    def _create_api(self, api_ver=3):
         url = 'https://%s/ovirt-engine/api' % self.ip()
-        return ovirtsdk.api.API(
-            url=url,
-            username=constants.ENGINE_USER,
-            password=str(self.metadata['ovirt-engine-password']),
-            validate_cert_chain=False,
-            insecure=True,
-        )
+        if api_ver == 3:
+            return ovirtsdk.api.API(
+                url=url,
+                username=constants.ENGINE_USER,
+                password=str(self.metadata['ovirt-engine-password']),
+                validate_cert_chain=False,
+                insecure=True,
+            )
 
-    def _get_api(self):
+    def _get_api_v3(self):
```

Comment:
This does not need to be renamed at all no? just add the api_ver param and pass it to the _create_api, and then this can be reused for any version.